controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -659,12 +659,23 @@ bool master_thread::autoconfig_wsc_add_m2(WSC::m1 &m1,
         cfg.auth_type   = bss_info_conf->authentication_type;
         cfg.encr_type   = bss_info_conf->encryption_type;
         cfg.network_key = bss_info_conf->network_key;
-        cfg.bss_type    = bss_info_conf->bss_type;
+        // Since currently we do not support hybrid mode, the options are either fronthaul,
+        // backhaul, or teardown.
+        if (bss_info_conf->fronthaul) {
+            cfg.bss_type = WSC::eWscVendorExtSubelementBssType::FRONTHAUL_BSS;
+        } else if (bss_info_conf->backhaul) {
+            cfg.bss_type = WSC::eWscVendorExtSubelementBssType::BACKHAUL_BSS;
+        } else {
+            cfg.bss_type = WSC::eWscVendorExtSubelementBssType::TEARDOWN;
+        }
```

Comment:
NACK this!
- We *never* should set TEARDOWN here, teardown is handled in the else branch below.
- We should simply set the hybrid mode if hybrid mode is requested. If we don't do this, controller certification *will* fail.

```suggestion
        cfg.bss_type = 0;
        if (bss_info_conf->fronthaul) {
            cfg.bss_type |= WSC::eWscVendorExtSubelementBssType::FRONTHAUL_BSS;
        }
        if (bss_info_conf->backhaul) {
            cfg.bss_type |= WSC::eWscVendorExtSubelementBssType::BACKHAUL_BSS;
        }
```

Reply:
You're right, fixed.